pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -821,17 +819,12 @@ func GetSelectorRegexMap(l7 policy.L7DataMap, lru *lru.Cache) (CachedSelectorREE
 			}
 		}
 		mp := strings.Join(reStrings, "|")
-		rei, ok := lru.Get(mp)
-		if ok {
-			re := rei.(*regexp.Regexp)
-			newRE[selector] = re
-		} else {
-			re, err := regexp.Compile(mp)
-			if err != nil {
-				return nil, err
-			}
-			lru.Add(mp, re)
-			newRE[selector] = re
+		rei, err := re.CompileRegex(mp)
+		switch {
+		case rei != nil && err == nil:
+			newRE[selector] = rei
+		case err != nil:
+			return nil, err
```

Comment:
Any particular reason this doesn't follow the typical go form?

```suggestion
		rei, err := re.CompileRegex(mp)
		if err {
			return nil, err
		}
		newRE[selector] = rei
```

Reply:
No, probably a leftover from iterating on the design of `CompileRegex`.